cypress/integration/dimensions/period.cy.js

```diff
@@ -0,0 +1,484 @@
+            { name: 'Bi-weekly', amountOfChildren: 26 },
+            { name: 'Monthly', amountOfChildren: 12 },
+            { name: 'Bi-monthly', amountOfChildren: 6 },
+            { name: 'Quarterly', amountOfChildren: 4 },
+            { name: 'Six-monthly', amountOfChildren: 2 },
+            { name: 'Six-monthly April', amountOfChildren: 2 },
+            { name: 'Yearly', amountOfChildren: 10 },
+            { name: 'Financial year (Start November)', amountOfChildren: 10 },
+            { name: 'Financial year (Start October)', amountOfChildren: 10 },
+            { name: 'Financial year (Start July)', amountOfChildren: 10 },
+            { name: 'Financial year (Start April)', amountOfChildren: 10 },
+        ]
+        it('can switch to fixed periods', () => {
+            switchToFixedPeriods()
+        })
+        it("period type is 'Monthly'", () => {
+            expectFixedPeriodTypeToBe('Monthly')
+        })
+        it('has the correct period types', () => {
+            openFixedPeriodsTypeSelect()
+            fixedPeriodTypes.forEach(type =>
+                expectFixedPeriodTypeSelectToContain(type.name)
+            )
+            selectPeriodType(fixedPeriodTypes[1].name) // Click the second item to close the dropdown
+        })
+        fixedPeriodTypes.forEach(type =>
+            it(`'${type.name}' has ${type.amountOfChildren} items`, () => {
+                switchFixedPeriodType(type.name)
+                type.amountOfChildren > 50
+                    ? expectSelectablePeriodsAmountToBeLeast(
+                          type.amountOfChildren
+                      )
+                    : expectSelectablePeriodsAmountToBe(type.amountOfChildren)
+            })
+        )
+    })
+    describe('period settings - hidden monthly', () => {
+        it('navigates to the start page', () => {
+            cy.intercept(
+                /systemSettings(\S)*keyAnalysisRelativePeriod(\S)*/,
+                req => {
+                    req.reply(res => {
+                        res.send({
+                            body: {
+                                ...systemSettingsBody,
+                                keyHideMonthlyPeriods: true,
+                            },
+                        })
+                    })
+                }
+            )
+            goToStartPage()
+        })
+        it('opens the period dimension modal', () => {
+            openDimension(DIMENSION_ID_PERIOD)
+            expectPeriodDimensionModalToBeVisible()
+        })
+        it("period type is 'Quarters'", () => {
+            expectRelativePeriodTypeToBe('Quarters')
+        })
+        it("relative option 'Months' is not shown", () => {
+            openRelativePeriodsTypeSelect()
+            expectRelativePeriodTypeSelectToNotContain('Months')
+        })
+        it('all other relative options are shown', () => {
+            const relativePeriodTypes = [
```

Comment:
I guess this is where you talking about refactoring? Yeah, I agree there is a lot of repetition. What about having a master array `allRelativePeriods` and `allFixedPeriods`, and then doing something like: 
```
const relativePeriodsWithoutMonthly = allRelativePeriods.filter(p => p !== 'Monthly')
```

Then I guess you would have to create the object for the amountOfChildren tests from the `allRelativePeriods` array.

Reply:
Yeah I considered it, but I think the tests are already too dynamic as they are. Best to keep it simple and explicitly provide the expect result so we can be sure that the test works. 
However I did change a lot of the naming of props and variables to be easier to read and follow. Hopefully that solves a bit of the confusion at least 🤞